kokoro/scripts/test/go_test.sh

```diff
@@ -64,6 +64,15 @@ else:
  print(data)"
 }
 
+# A helper function for joining a bash array.
+# Ex. join_by , a b c -> a,b,c
+function join_by {
+  local d=${1-} f=${2-}
+  if shift 2; then
+    printf %s "$f" "${@/#/$d}"
```

Comment:
I found this very hard to understand. At the very least this needds comments and d/f need to be renamed.

I had to read this whole article to get what it's doing:
https://dev.to/meleu/how-to-join-array-elements-in-a-bash-script-303a

Plus it doesn't work in the admittedly strange case of having a `&` in the delimiter. Which I discovered only after reading all of the comments on https://stackoverflow.com/a/17841619. This gotcha reinforces my feeling that this is using too much bash magic.

How about:
```
function join_by() {
  delim="$1"
  for (( i = 2; i <= $#; i++)); do
    printf "${!i}"  # The ith positional argument
    if [[ $i -ne $# ]]; then
      printf "$delim"
    fi
  done
}
```

Reply:
Agreed on it being hard to understand. Used your version.